packages/trader/src/Modules/Trading/Components/Form/WheelPicker/wheel-picker-mobile.tsx

```diff
@@ -0,0 +1,168 @@
+    const snapY = Number(yLeftover >= snapThreshold);
+
+    const absDeltaIndex = Math.floor(absDeltaY / optionHeight) + snapY;
+    const deltaIndex = absDeltaIndex * Math.sign(deltaY) * -1;
+
+    const MIN_INDEX = 0;
+    const MAX_INDEX = options.length - 1;
+    const targetIndex = Math.min(MAX_INDEX, Math.max(MIN_INDEX, selectedIndex + deltaIndex));
+    return targetIndex;
+}
+
+const WheelPickerMobile: React.FC<WheelPickerMobileProps> = ({
+    defaultValue,
+    onChange,
+    currency,
+    options,
+    optionHeight = 28,
+}) => {
+    const [selectedIndex, setSelectedIndex] = useState<number>(options.length - 1);
+    const [optionWidth, setOptionWidth] = useState<number>(0);
+
+    const [swipe, setSwipe] = useState<SwipeState>({ startY: 0, deltaY: 0, translateY: 0 });
+    const optionRef = useRef<HTMLDivElement>(null);
+
+    const swipeableHandlers = useSwipeable({
+        onSwipeStart: () => {
+            setSwipe(swipe => ({ ...swipe, startY: swipe.translateY }));
+        },
+        onSwiping: ({ deltaY, first }) => {
+            if (first) return;
+            const MIN = 0;
+            const MAX = optionHeight * (options.length - 1);
+            const translateY = Math.min(MAX, Math.max(MIN, swipe.startY + deltaY));
+            const newIndex = Math.min(options.length - 1, Math.max(0, Math.round(translateY / optionHeight)));
+            setSwipe(swipe => ({
+                ...swipe,
+                deltaY,
+                translateY: (translateY + optionHeight * newIndex) / 2,
+            }));
+            setSelectedIndex(newIndex);
+        },
+        onSwiped: ({ deltaY }) => {
+            const MIN = 0;
+            const MAX = optionHeight * (options.length - 1);
+            const translateY = Math.min(MAX, Math.max(MIN, swipe.startY + deltaY));
+            const newIndex = Math.min(options.length - 1, Math.max(0, Math.round(translateY / optionHeight)));
+            setSwipe(swipe => ({ ...swipe, deltaY: 0, translateY: optionHeight * newIndex }));
+        },
+        trackMouse: true,
+        delta: optionHeight,
+    });
+
+    useEffect(() => {
+        const defaultValueIndex = options.findIndex(o => o === defaultValue);
+        setSelectedIndex(defaultValueIndex !== -1 ? defaultValueIndex : options.length - 1);
+    }, []);
+
+    useEffect(() => {
+        const translateY = optionHeight * selectedIndex;
+        setSwipe(swipe => ({ ...swipe, translateY }));
+    }, [selectedIndex, options, optionHeight]);
+
+    useEffect(() => {
+        if (onChange) onChange(options[selectedIndex]);
+        if (optionRef.current) {
+            setOptionWidth(optionRef.current.offsetWidth);
+        }
+    }, [selectedIndex, onChange, options]);
+
+    const handleClick = (index: number) => {
+        setSelectedIndex(index);
+    };
+
+    const translateValue = `translateY(${-swipe.translateY}px)`;
+    const wheelStyle = {
+        transform: translateValue,
```

Comment:
Can't we move some of these to the stylesheet wherever possible?

Reply:
The Style is created dynamically.  transform and top properties  are getting created dynamically as per the swipe and optionHeight . Not sure where we can move that